tests/performance/profiling.py

```diff
@@ -0,0 +1,77 @@
+import timeit
+import tracemalloc
+from multiprocessing import Process, Value
+
+import pandas as pd
+
+
+def _profile_time(method, dataset, iterations=100):
+    total_time = 0
+    for _ in range(iterations):
+        start_time = timeit.default_timer()
+        method(dataset)
+        total_time += timeit.default_timer() - start_time
+
+    return total_time / iterations
+
+
+def _set_memory_for_method(method, dataset, peak_memory):
+    tracemalloc.start()
+    method(dataset)
+    peak_memory.value = tracemalloc.get_traced_memory()[1]
+    tracemalloc.stop()
+
+
+def _profile_memory(method, dataset):
+    peak_memory = Value('i', 0)
+    profiling_process = Process(target=_set_memory_for_method, args=(method, dataset, peak_memory))
+    profiling_process.start()
+    profiling_process.join()
+    return peak_memory.value
+
+
+def profile_transformer(transformer, dataset_generator, transform_size, fit_size=None):
+    """
+    Function to profile a transformer.
```

Comment:
This should be on the line above. I also suspect that `lint` may not be checking this folder properly, since I think this should be reported by `pylint`

Reply:
The linter definitely complains about this file when I do certain things. Idk why that wasn't reported